concrete-core-fixture/src/fixture/lwe_ciphertext_creation.rs

```diff
@@ -0,0 +1,119 @@
+
+    fn generate_parameters_iterator() -> Box<dyn Iterator<Item = Self::Parameters>> {
+        Box::new(
+            vec![
+                LweCiphertextCreationParameters {
+                    lwe_dimension: LweDimension(1),
+                },
+                LweCiphertextCreationParameters {
+                    lwe_dimension: LweDimension(512),
+                },
+                LweCiphertextCreationParameters {
+                    lwe_dimension: LweDimension(751),
+                },
+            ]
+            .into_iter(),
+        )
+    }
+
+    fn generate_random_repetition_prototypes(
+        _parameters: &Self::Parameters,
+        _maker: &mut Maker,
+    ) -> Self::RepetitionPrototypes {
+    }
+
+    fn generate_random_sample_prototypes(
+        parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+    ) -> Self::SamplePrototypes {
+        let num_elements = parameters.lwe_dimension.to_lwe_size().0;
+        (maker.transform_raw_to_container(&Precision::Raw::uniform_vec(num_elements)),)
+    }
+
+    fn prepare_context(
+        _parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+        sample_proto: &Self::SamplePrototypes,
+    ) -> Self::PreExecutionContext {
+        (maker.synthesize_container(&sample_proto.0),)
+    }
+
+    fn execute_engine(
+        _parameters: &Self::Parameters,
+        engine: &mut Engine,
+        context: Self::PreExecutionContext,
+    ) -> Self::PostExecutionContext {
+        let (underlying_container,) = context;
+        let lwe_ciphertext =
+            unsafe { engine.create_lwe_ciphertext_unchecked(underlying_container) };
+        (lwe_ciphertext,)
+    }
+
+    fn process_context(
+        _parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+        sample_proto: &Self::SamplePrototypes,
+        context: Self::PostExecutionContext,
+    ) -> Self::Outcome {
+        let (lwe_ciphertext,) = context;
+        let ciphertext_proto = maker.unsynthesize_lwe_ciphertext(lwe_ciphertext);
+        (
+            maker.transform_container_to_raw(&sample_proto.0),
+            maker.transform_ciphertext_to_raw(&ciphertext_proto),
+        )
+    }
+
+    fn compute_criteria(
+        _parameters: &Self::Parameters,
+        _maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+    ) -> Self::Criteria {
+        (Variance(0.),)
+    }
+
+    fn verify(criteria: &Self::Criteria, outputs: &[Self::Outcome]) -> bool {
+        let (means, actual): (Vec<_>, Vec<_>) = outputs.iter().cloned().unzip();
+        let means: Vec<Precision::Raw> = means.into_iter().flatten().collect();
+        let actual: Vec<Precision::Raw> = actual.into_iter().flatten().collect();
+        assert_noise_distribution(actual.as_slice(), means.as_slice(), criteria.0)
+    }
+}
```

Comment:
For this test, I would not even rely on the `assert_noise_distribution`, but on a plain `assert_eq` instead. In this case, you don't need to pass a variance as parameter. I think this remark goes for all the other fixtures.

Reply:
I think I saw this being used in other fixtures for retrieval over plaintext vectors I think?

Reply:
as discussed added https://github.com/zama-ai/concrete_internal/issues/449 to update fixtures that used a similar pattern which should probably use an `assert_eq`